src/inmanta/types.py

```diff
@@ -33,6 +33,15 @@
     from inmanta.protocol.common import ReturnValue  # noqa: F401
 
 
+# Typing of dataclass.* methods relies entirely on the definition in typeshed that only exists during typechecking.
+# This ensures that our code works during typechecking and at runtime.
+if not TYPE_CHECKING:
+    DataclassProtocol = object
+else:
+    import _typeshed
```

Comment:
Drawing attention to [this comment](https://github.com/inmanta/inmanta-core/pull/8399/files#r1928841910) once more (it gets lost in GitHub's thread collapsing).

Reply:
What is the actionable point?

Reply:
Make sure we either include typeshed as a dev dependency, or that we use the typeshed that's vendored in mypy.